src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -166,18 +150,20 @@ bool CanFlushUsingMembarrier()
     }
 #endif
 
+#if HAVE_SYS_MEMBARRIER_H
```

Comment:
My Ubuntu 24.04 doesn't have sys/membarrier.h, it has /usr/include/linux/membarrier.h only. So it seems this code path would never be built there.